chore: Add policy documentation to website (#5296)
cloudquery/cloudquery: website/pages/docs/plugins/sources/aws/policies.md

> # Policies and Compliance Frameworks

Reviewer: We should exclude this file via https://github.com/cloudquery/cloudquery/blob/812241697c644bdb1ae202bbadcb3baae456f788/.github/workflows/lint_markdown.yml#L19

> CloudQuery SQL Policies for AWS. See the [Readme on Github](https://github.com/cloudquery/cloudquery/tree/main/plugins/source/aws/policies) for installation instructions.

Reviewer: ```suggestion
CloudQuery SQL Policies for AWS. See the [Readme on GitHub](https://github.com/cloudquery/cloudquery/tree/main/plugins/source/aws/policies) for installation instructions.
```

  - Ensure MFA is enabled for all IAM users that have a console password (Scored)
  - Ensure credentials unused for 90 days or greater are disabled (Scored)
  - Ensure access keys are rotated every 90 days or less
  - Ensure IAM password policy requires at least one uppercase letter
  - Ensure IAM password policy requires at least one lowercase letter
  - Ensure IAM password policy requires at least one symbol
  - Ensure IAM password policy requires at least one number
  - Ensure IAM password policy requires minimum length of 14 or greater
  - Ensure IAM password policy prevents password reuse
  - Ensure IAM password policy expires passwords within 90 days or less
  - Ensure no root account access key exists (Scored)
  - Ensure MFA is enabled for the "root" account
  - Ensure hardware MFA is enabled for the "root" account (Scored)
  - IAM users should not have IAM policies attached
  - Ensure CloudTrail is enabled in all regions
  - Ensure CloudTrail log file validation is enabled
  - CloudTrail trails should be integrated with CloudWatch Logs
  - Ensure S3 bucket access logging is enabled on the CloudTrail S3 bucket
  - CloudTrail should have encryption at rest enabled

>   - Ensure rotation for customer created CMKs is enabled (Scored)

Reviewer: ```suggestion
  - Ensure rotation for customer created custom master keys is enabled (Scored)
```

Maybe it's better to use the full term instead of the acronym

  - Elasticsearch domains should be in a VPC
  - Elasticsearch domains should encrypt data sent between nodes
  - Elasticsearch domain error logging to CloudWatch Logs should be enabled
  - Elasticsearch domains should have audit logging enabled
  - Elasticsearch domains should have at least three data nodes
  - Elasticsearch domains should be configured with at least three dedicated master nodes
  - Connections to Elasticsearch domains should be encrypted using TLS 1.2
  - Classic Load Balancers with SSL/HTTPS listeners should use a certificate provided by AWS Certificate Manager
  - Classic Load Balancer listeners should be configured with HTTPS or TLS termination
  - Application load balancers should be configured to drop HTTP headers
  - Application and Classic Load Balancers logging should be enabled
  - Application Load Balancer deletion protection should be enabled
  - Classic Load Balancers should have connection draining enabled
  - Classic Load Balancers with HTTPS/SSL listeners should use a predefined security policy that has strong configuration
  - Application Load Balancer should be configured to redirect all HTTP requests to HTTPS

>   - EMR clusters should not have public ip addresses

Reviewer: ```suggestion
  - EMR clusters should not have public IP addresses
```

## AWS Public Egress

### Requirements
AWS Public Egress requires the following tables to be synced before the policy is executed:

```
tables:
  - aws_ec2_instances
  - aws_ec2_route_tables
  - aws_lambda_functions
```

### Queries
AWS Public Egress performs the following checks:
  - Find all ec2 instances that have unrestricted access to the internet with a wide open security group and routing
  - All ec2 instances that have unrestricted access to the internet via a security group
  - Find all lambda functions that have unrestricted access to the internet
## AWS Publicly Available

### Requirements
AWS Publicly Available requires the following tables to be synced before the policy is executed:

```
tables:
  - aws_apigateway_rest_apis
  - aws_apigatewayv2_apis
  - aws_cloudfront_distributions
  - aws_ec2_instances
  - aws_elbv1_load_balancers
  - aws_elbv2_load_balancers
  - aws_rds_instances
  - aws_redshift_clusters
```

### Queries
AWS Publicly Available performs the following checks:
  - Find all API Gateway instances that are publicly accessible
  - Find all API Gateway V2 instances (HTTP and Webhook) that are publicly accessible

>   - Find all Cloudfront distributions

Reviewer: ```suggestion
  - Find all CloudFront distributions
```

Not sure where this comes from but it should be CloudFront

```
tables:
  - aws_acm_certificates
  - aws_apigateway_api_keys
  - aws_backup_vault_recovery_points
  - aws_backup_vaults
  - aws_cloudfront_distributions
  - aws_cloudwatch_alarms
  - aws_directconnect_connections
  - aws_directconnect_lags
  - aws_dynamodb_tables
  - aws_ec2_ebs_volumes
  - aws_ec2_eips
  - aws_ec2_hosts
  - aws_ec2_images
  - aws_ec2_instances
  - aws_ec2_internet_gateways
  - aws_ec2_network_acls
  - aws_ec2_route_tables
  - aws_ec2_security_groups
  - aws_ec2_transit_gateway_attachments
  - aws_ec2_transit_gateways
  - aws_ecr_repositories
  - aws_ecr_repository_images
  - aws_efs_filesystems
  - aws_elbv2_listeners
  - aws_elbv2_load_balancers
  - aws_elbv2_target_groups
  - aws_lightsail_container_service_deployments
  - aws_lightsail_container_services
  - aws_lightsail_disks
  - aws_lightsail_distributions
  - aws_lightsail_load_balancers
  - aws_lightsail_static_ips
  - aws_route53_hosted_zones
  - aws_sns_subscriptions
  - aws_sns_topics
```

### Queries
AWS Unused Resources performs the following checks:
  - Unused ACM certificate
  - Unused API Gateway API key
  - Vaults with no recovery points
  - Disabled CloudFront distribution

>   - Disabled Cloudwatch alarm

Reviewer: ```suggestion
  - Disabled CloudWatch alarm
```
